src/java/edu/ucsd/library/dams/file/characterization/processors/ChecksumProcessor.java

```diff
@@ -0,0 +1,56 @@
+package edu.ucsd.library.dams.file.characterization.processors;
+
+import static edu.ucsd.library.dams.file.characterization.model.MetadataConstants.CHECKSUM_CRC32;
+import static edu.ucsd.library.dams.file.characterization.model.MetadataConstants.CHECKSUM_MD5;
+import static edu.ucsd.library.dams.file.characterization.model.MetadataConstants.CHECKSUM_SHA;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.NoSuchAlgorithmException;
+import java.util.HashMap;
+import java.util.Map;
+
+import edu.ucsd.library.dams.file.Checksum;
+
+/**
+ * Calculate checksums of a file.
+ * @author lsitu@ucsd.edu
+**/
+public class ChecksumProcessor extends Processor {
+
+    public ChecksumProcessor() {
+        super(null);
+    }
+
+    @Override
+    public Map<String, Object> extractMetadata(String sourceFile) throws Exception {
+        return process(sourceFile);
+    }
+
+    /*
+     * Calculate the checksums with list of digest algorithms by piping it through a DigestInputStream
+     * @param sourceFile the path to the file
+     * @param algorithms the digest algorithms
+     * @return TechnicalMetadata
+     * @throws 
+     */
+    private Map<String, Object> process(String sourceFile)
+            throws FileNotFoundException, IOException, NoSuchAlgorithmException {
+
+        Map<String, Object> metadata = new HashMap<>();
+
+        try (InputStream in = new FileInputStream(sourceFile)) {
+
+            boolean crc32Checksum = true;
+            boolean md5Checksum = true;
+            boolean sha1Checksum = true;
+            Map<String, String> checksums = Checksum.checksums(in, null, crc32Checksum, md5Checksum, sha1Checksum, false, false);
+            metadata.put(CHECKSUM_CRC32, checksums.get("crc32"));
+            metadata.put(CHECKSUM_MD5, checksums.get("md5"));
+            metadata.put(CHECKSUM_SHA, checksums.get("sha1"));
```

Comment:
Ah, nevermind, I see there's a sha1 hash defined here. Since there are various sha checkums, perhaps the constant should be `CHECKSUM_SHA1`?

Reply:
@mcritchlow Yeah, Good question. The term `sha` for `sha1` is borrowed from the former Jhove process to make it consistent. And also, in Fedora now `sha` means the default `sha1` while handling Want-Digest checksums . But we can use `CHECKSUM_SHA1`. Would you like me to make the change now?

Reply:
Okay, I just renamed constant `CHECKSUM_SHA`  to `CHECKSUM_SHA1` anyway. See commit https://github.com/ucsdlib/damsrepo/pull/68/commits/f8bfec2358681f12b8080660139c9f1a8ee50a2e.

Reply:
Yeah I think that'll be nice, thanks. Going forward, we may want to eventually switch to 256 or something, so having the constant spell out the digest explicitly will be a nice easy reference :+1: